src/test/java/org/broadinstitute/hellbender/utils/io/IOUtilsUnitTest.java

```diff
@@ -270,5 +267,44 @@ public void testUrlEncodeDecode(final String string, final String encoded) {
         Assert.assertEquals(string, IOUtils.urlDecode(encoded));
     }
 
+    @DataProvider(name="resourcePaths")
+    public Object[][] getResourcePaths() {
+        final String testResourceContents = "this is a test resource file";
+        return new Object[][] {
+                { Resource.LARGE_RUNTIME_RESOURCES_PATH + "/testResourceFile.txt", null , testResourceContents},
+                { "testResourceFile.txt", IOUtils.class , testResourceContents},
+        };
+    }
+
+    @Test(dataProvider = "resourcePaths")
+    public void testWriteTempResource(
+            final String resourcePath, final Class<?> relativeClass, final String expectedFirstLine) throws IOException
+    {
+        final Resource largeResource = new Resource(resourcePath, relativeClass);
+        final File resourceFile = IOUtils.writeTempResource(largeResource);
+        final String resourceContentsFirstLine = getFirstLineAndDeleteTempFile(resourceFile);
```

Comment:
Should the `IOUtils.writeTempResource()` methods set the extracted resource to be deleted on JVM exit? It seems like we will end up leaving garbage around if we don't do that in the methods themselves.

Reply:
We probably should. I'm going to do that in a separate PR though since all of the call sites should be cleaned up as well.